--- a/GTG/tools/networkmanager.py	2019-06-28 17:44:45.788942062 +0200
+++ b/GTG/tools/networkmanager.py	2019-06-28 17:44:50.627891003 +0200
@@ -20,13 +20,12 @@
 
 """ Communicate with Network Manager """
 
-from gi.repository import NetworkManager, NMClient
-
+from gi.repository import Gio
 
 def is_connection_up():
     """ Returns True if GTG can access the Internet """
-    state = NMClient.Client().get_state()
-    return state == NetworkManager.State.CONNECTED_GLOBAL
+    network_monitor = Gio.NetworkMonitor.get_default()
+    return network_monitor.get_network_available()
 